core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -698,9 +698,10 @@ public MergeWriterNode plan(Merge merge)
                 int index = mergeCaseSetColumns.indexOf(dataColumnHandle);
                 if (index >= 0) {
                     Expression original = mergeCase.getSetExpressions().get(index);
-                    Expression setExpression = coerceIfNecessary(analysis, original, original);
+                    Expression setExpression = original;
```

Comment:
This variable doesn't seem to be needed anymore. You could rename `original` to `setExpression` if that is more clear.

Reply:
Removed/renamed.